src/cclib/parser/qchemparser.py

```diff
@@ -309,45 +385,142 @@ def extract(self, inputfile, line):
+                                # then it indexes (1-based) that
+                                # specific atom in the whole molecule.
+                                tokens = line.split()
+                                assert len(tokens) > 0
+                                element = tokens[0][0].upper() + tokens[0][1:].lower()
+                                assert element in self.table.element
+                                if len(tokens) > 1:
+                                    assert len(tokens) == 2
+                                    index = int(tokens[1]) - 1
+                                else:
+                                    index = -1
+                                line = next(inputfile)
+
+                                # Next comes the ECP definition. If
+                                # the line contains only a single
+                                # item, it's a built-in ECP, otherwise
+                                # it's a full definition.
+                                tokens = line.split()
+                                if len(tokens) == 1:
+                                    ecp_name = tokens[0]
+                                    ncore = 0
+                                    line = next(inputfile)
+                                else:
+                                    assert len(tokens) == 3
+                                    ecp_name = tokens[0]
+                                    maxl = int(tokens[1])
+                                    ncore = int(tokens[2])
+                                    # Don't parse the remainder of the
+                                    # ECP definition.
+                                    while list(set(line.strip())) != ['*']:
+                                        line = next(inputfile)
+
+                                entry = (element, index, ncore)
+                                self.user_input['ecp'].append(entry)
+
+                                line = next(inputfile)
+
+                                if line.strip().lower() == '$end':
+                                    break
+
+                    if line.strip().lower() == '$molecule':
+
+                        self.user_input['molecule'] = dict()
+                        line = next(inputfile)
+
+                        # Don't read the molecule, only the
+                        # supersystem charge and multiplicity.
+                        if line.split()[0].lower() == 'read':
+                            pass
+                        else:
+                            charge, mult = [int(x) for x in line.split()]
+                            self.user_input['molecule']['charge'] = charge
+                            self.user_input['molecule']['mult'] = mult
+
+                    line = next(inputfile).lower()
 
             # Parse the basis set name
             if 'Requested basis set' in line:
+                # if 'is non-standard' not in line
```

Comment:
Commented-out code that should be removed?

Reply:
A fix for another day...